docs/sources/tempo/setup/operator/object-storage.md

```diff
@@ -0,0 +1,175 @@
+
+## AWS S3
+
+### Requirements
+
+* Create a [bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html) on AWS.
+
+### Installation
+
+* Deploy the Tempo Operator to your cluster.
+
+* Create an Object Storage secret with keys as follows:
+
+    ```console
+    kubectl create secret generic tempostack-dev-s3 \
+      --from-literal=bucket="<BUCKET_NAME>" \
+      --from-literal=endpoint="<AWS_BUCKET_ENDPOINT>" \
+      --from-literal=access_key_id="<AWS_ACCESS_KEY_ID>" \
+      --from-literal=access_key_secret="<AWS_ACCESS_KEY_SECRET>"
+    ```
+
+  where `tempostack-dev-s3` is the secret name.
+
+* Create an instance of TempoStack by referencing the secret name and type as `s3`:
+
+  ```yaml
+  spec:
+    storage:
+      secret:
+        name: tempostack-dev-s3
+        type: s3
+  ```
+
+## Azure
+
+### Requirements
+
+* Create a [bucket](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction) on Azure.
+
+### Installation
+
+* Deploy the Tempo Operator to your cluster.
+
+* Create an Object Storage secret with keys as follows:
+
+    ```console
+    kubectl create secret generic tempostack-dev-azure \
+      --from-literal=container="<AZURE_CONTAINER_NAME>" \
+      --from-literal=account_name="<AZURE_ACCOUNT_NAME>" \
+      --from-literal=account_key="<AZURE_ACCOUNT_KEY>"
+    ```
+
+  where `tempostack-dev-azure` is the secret name.
+
+* Create an instance of TempoStack by referencing the secret name and type as `azure`:
+
+  ```yaml
+  spec:
+    storage:
+      secret:
+        name: tempostack-dev-azure
+        type: azure
+  ```
+
+## Google Cloud Storage
+
+### Requirements
+
+* Create a [project](https://cloud.google.com/resource-manager/docs/creating-managing-projects) on Google Cloud Platform.
+* Create a [bucket](https://cloud.google.com/storage/docs/creating-buckets) under same project.
+* Create a [service account](https://cloud.google.com/docs/authentication/getting-started#creating_a_service_account) under same project for GCP authentication.
+
+### Installation
+
+* Deploy the Tempo Operator to your cluster.
+
+* Copy the service account credentials received from GCP into a file name `key.json`.
+
+* Create an Object Storage secret with keys `bucketname` and `key.json` as follows:
+
+    ```console
+    kubectl create secret generic tempostack-dev-gcs \
+      --from-literal=bucketname="<BUCKET_NAME>" \
+      --from-file=key.json="<PATH/TO/KEY.JSON>"
+    ```
+
+  where `tempostack-dev-gcs` is the secret name, `<BUCKET_NAME>` is the name of bucket created in requirements step and `<PATH/TO/KEY.JSON>` is the file path where the `key.json` was copied to.
+
+* Create an instance of TempoStack by referencing the secret name and type as `gcs`:
+
+  ```yaml
+  spec:
+    storage:
+      secret:
+        name: tempostack-dev-gcs
+        type: gcs
+  ```
+
+## Minio
```

Comment:
According to the MinIO website (linked below), the capitalization should be MinIO. 
```suggestion
## MinIO
```